any_parser/any_parser.py

```diff
@@ -3,254 +3,209 @@
+            url (str): The URL of the AnyParser RT API.
 
         Returns:
-            str: The file id of the uploaded file.
+            None
         """
-        file_extension = Path(file_path).suffix.lower().lstrip(".")
-
-        # Check if the file exists and file_type
-        error = check_file_type_and_path(file_path, file_extension)
-        if error:
-            return error
-
-        file_name = Path(file_path).name
-
-        # Create the JSON payload
-        payload = {
-            "file_name": file_name,
-            "process_type": process_type.value,
-        }
-
-        if extract_args is not None and isinstance(extract_args, dict):
-            payload["extract_args"] = extract_args
-
-        # Send the POST request
-        response = requests.post(
-            self._async_upload_url,
-            headers=self._headers,
-            data=json.dumps(payload),
-            timeout=TIMEOUT,
-        )
-
-        # If response successful, upload the file
-        return upload_file_to_presigned_url(file_path, response)
+        self._sync_parser = SyncParser(api_key, base_url)
+        self._async_parser = AsyncParser(api_key, base_url)
 
+    @handle_parsing
     def parse(
         self,
-        file_path: str,
-        extract_args: Optional[Dict] = None,
-    ) -> Tuple[str, str]:
-        """Extract full content from a file in real-time.
-
-        Args:
-            file_path (str): The path to the file to be parsed.
-            extract_args (Optional[Dict]): Additional extraction arguments added
-                to the prompt.
-
-        Returns:
-            tuple(str, str): The extracted data and the time taken.
-        """
-
-        response, info = self._get_sync_response(
-            self._sync_parse_url,
-            file_path,
+        file_path=None,
+        file_content=None,
+        file_type=None,
+        extract_args=None,
+    ):
+        """Extract full content from a file in real-time."""
+        response, info = self._sync_parser.get_sync_response(
+            self._sync_parser._sync_parse_url,
+            file_content=file_content,  # type: ignore
+            file_type=file_type,  # type: ignore
             extract_args=extract_args,
         )
+
         if response is None:
             return info, ""
 
         try:
             response_data = response.json()
-            response_list = []
-            for text in response_data["markdown"]:
-                response_list.append(text)
-            markdown_text = "\n".join(response_list)
-            return (
-                markdown_text,
-                f"Time Elapsed: {info}",
-            )
+            result = "\n".join(
+                response_data["markdown"]
+            )  # Using direct extraction instead of extract_key
+            return result, f"Time Elapsed: {info}"
         except json.JSONDecodeError:
             return f"Error: Invalid JSON response: {response.text}", ""
 
+    @handle_parsing
```

Comment:
nit: as I stated above, this is the problem to use a parsing decorator for extract. This will make the code really confusing and hard to read, extend, and maintain in the future.